File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
using namespace std;

int cnt[200001];
int solution(vector<int> nums){
    int ans = nums.size()/2, category = 0;

    for(auto n : nums){
        if(cnt[n] == 0){
            category += 1;
        }
        cnt[n] = 1;
    }

    return ans < category ? ans : category;
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {

    
   public int solution(int[] nums) {
        int choice = nums.length/2;

        Set<Integer> st = new HashSet<>(Arrays.stream(nums)
                                            .boxed()
                                            .collect(Collectors.toList()));
        
        return Math.min(choice, st.size());

    }
}